lgi/marshal.c

```diff
@@ -112,7 +132,7 @@ marshal_2lua_int (lua_State *L, GITypeTag tag, GIArgument *val,
 	    ReturnUnion *ru = (ReturnUnion *) val;			\
 	    ru->arg.v_ ## namelower = (g ## namelower) ru->ut;		\
 	  }								\
-	lua_pushnumber (L, parent == LGI_PARENT_FORCE_POINTER		\
+	lua_pushinteger (L, parent == LGI_PARENT_FORCE_POINTER		\
```

Comment:
This change assumes that the Lua integer type has at least 64 bits. Actually... since `double` only has something like a 53 bits mantissa, I guess that (u)int64 passing never really worked correctly anyway, so this can be ignored. Right?